xbmc/video/VideoDatabase.cpp

```diff
@@ -3020,6 +3020,49 @@ int CVideoDatabase::SetDetailsForSeason(const CVideoInfoTag& details, const std:
   return -1;
 }
 
+bool CVideoDatabase::SetFileForEpisode(const std::string& fileAndPath, int idEpisode, int idFile)
+{
+  try
+  {
+    BeginTransaction();
+    std::string sql = PrepareSQL("UPDATE episode SET c18='%s', idFile=%i WHERE idEpisode=%i",
+                                 fileAndPath.c_str(), idFile, idEpisode);
+    m_pDS->exec(sql);
+    CommitTransaction();
+
+    return true;
+  }
+  catch (...)
+  {
+    CLog::Log(LOGERROR, "{} ({}) failed", __FUNCTION__, idEpisode);
+  }
+  RollbackTransaction();
+  return false;
+}
+
+bool CVideoDatabase::SetFileForMovie(const std::string& fileAndPath, int idMovie, int idFile)
+{
+  try
+  {
+    BeginTransaction();
+    std::string sql = PrepareSQL("UPDATE movie SET c22='%s', idFile=%i WHERE idMovie=%i",
+                                 fileAndPath.c_str(), idFile, idMovie);
+    m_pDS->exec(sql);
+    sql = PrepareSQL("UPDATE videoversion SET idFile=%i WHERE idMedia=%i AND media_type='movie'",
```

Comment:
Sorry for noticing this late and after a fixup commit was just merged.
This line is wrong, it will update all versions/extra of the movie and break them. videoversion.idFile is the only link between a parent movie and its additional versions/extras.
This also breaks version artwork, which uses idFile as key in the art table.
What is the need to change the idFile?
While this function may do the correct thing for a standalone/parent movie, it won't work if you try to apply this to an additional version or an extra.

Reply:
idFile is changed when the initial path to an ISO/BDMV is resolved to a bluray:// url by user selection. Off the top of my head I think I kept the original so it is not re-added when the directory changes and library is refreshed.

Reply:
Actually that line doesn't work if there is more than one version - is triggers and exceptoin - idFile is a primary key so can't appear twice.

```
    BeginTransaction();
    std::string sql = PrepareSQL("UPDATE movie SET c22='%s', idFile=%i WHERE idMovie=%i",
                                 fileAndPath.c_str(), idFile, idMovie);
    m_pDS->exec(sql);

    sql = PrepareSQL(
        "UPDATE videoversion SET idFile=%i WHERE idFile=%i AND idMedia=%i AND media_type='movie'",
        idFile, oldIdFile, idMovie);
    m_pDS->exec(sql);
    CommitTransaction();

        sql = PrepareSQL(
        "UPDATE art SET media_id=%i WHERE media_id=%i AND media_type='videoversion'",
        idFile, oldIdFile);
    m_pDS->exec(sql);
    CommitTransaction();
```

Seems to do the trick.

Reply:
That logic is OK to update the default version of the movie (though the videoversion and art update would be better located in a new SetFileForVideoAsset function). There is an extra CommitTransaction() to remove line 10.

I don't see how it could work fine for the non-default version of a movie. Let's say you have uhd bd iso of a movie as default version and the bd iso as secondary version, and you play the bd.
Modifying the movie record with the bd info would be wrong. Changing idFile changes the default version of the movie > not correct, Changing c22 (path?) seems wrong too - an official file structure hasn't been decided yet but it's unlikely that all versions of a movie will have to live in the same folder.

(to be verified, not related to version). Another potential problem with the idFile change could come from the library scanner / info refresh creating an additional copy of the movie in the library since there is no movie record matching the idFile of the file found by scan anymore.

Reply:
Thanks.
I hadn't realised that additional versions were removed from the movies database.
I have a couple of ideas and will generate another PR soon.